### ISSUE TITLE
Create Spring Properties and Configuration

### ISSUE THREAD

**Issue by rtisma** [overture-stack/SONG#493]
Spec: https://wiki.oicr.on.ca/pages/viewpage.action?pageId=128222479

This include updating the application.yml, creating the correct Properties class (for property binding), and Config class (for bean definition). Also, LocalIdService, FederatedService, StaticAuth, DynamicAuth need to be conditionally wired, based upon `isDefined` criteria for certain config fields (as depicted in spec). 

Also need to validate entity urlTemplates, only at boot, that they contain the right urlTemplateVariables

Depends on #489, #491, #492

**Comment by rtisma** [comment]
closing as already done in release 3.0.1
